cabal-install/Distribution/Client/Dependency/Modular/Explore.hs

```diff
@@ -5,18 +5,19 @@ module Distribution.Client.Dependency.Modular.Explore
 
 import Data.Foldable as F
 import Data.Map as M
-import Data.Set as S
 
 import Distribution.Client.Dependency.Modular.Assignment
 import Distribution.Client.Dependency.Modular.Dependency
 import Distribution.Client.Dependency.Modular.Log
 import Distribution.Client.Dependency.Modular.Message
 import Distribution.Client.Dependency.Modular.Package
 import qualified Distribution.Client.Dependency.Modular.PSQ as P
+import qualified Distribution.Client.Dependency.Modular.ConflictSet as CS
 import Distribution.Client.Dependency.Modular.Tree
 import qualified Distribution.Client.Dependency.Types as T
 
--- | This function takes the variable we're currently considering and a
+-- | This function takes the variable we're currently considering, an
+-- initial conflict set (the goal reason for the current goal) and a
```

Comment:
I think it would be nice to include an explanation of how this corresponds to a `_|_` choice, like in https://github.com/haskell/cabal/pull/3357#issuecomment-212298258.  That definitely made it clearer for me.